### PR TITLE
Spec cleanup

### PR DESCRIPTION
**Pull request by bdunne**
Ensure collections are built with the correct type.

Fix lots of spec failures due to this change.
Add FactoryGirl and reduce duplication through factories and shared_examples

### CODE REVIEW

**Comment by bdunne** [comment]
@syncrou Please review.
cc @gmcculloug 
+ @jrafanie for the :scissors: :fire: :smile:

File: spec/group_spec.rb [hunk 1]
```diff
@@ -1,50 +1,17 @@
...
-                   :url => '/api/v1/groups/1/', :name => 'test1'},
-                  {:id => 2, :type => 'groups',
-                   :url => '/api/v1/groups/2/', :name => 'test2'}]}.to_json
-  end
-
-  let(:one_result) do
-    {:id => 1, :url => '/api/v1/groups/1/', :name => 'test1'}.to_json
-  end
-
   let(:api_connection) { instance_double("Faraday::Connection", :get => get) }
```
Comment (syncrou) [inline spec/group_spec.rb]: Do we need the get method as part of this instance_double? (on any of the specs that don't actually call ':get')

**Comment by syncrou** [comment]
:+1:  - minus the ':get' question.

**Comment by bdunne** [comment]
The `include_examples "Collection Methods"` still depends on that.

**Pull request merged**